File: app/src/main/java/com/scop/org/minesweeper/GameActivity.java
package com.scop.org.minesweeper;

import android.app.Activity;
import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.Window;
import android.view.WindowManager;

import com.scop.org.minesweeper.control.ScreenProperties;
import com.scop.org.minesweeper.elements.Grid;


/**
 * An example full-screen activity that shows and hides the system UI (i.e.
 * status bar and navigation/system bar) with user interaction.
 */
public class GameActivity extends Activity {
    private GamePanel gamepanel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);

        Bundle extras = this.getIntent().getExtras();
        if (extras==null){
            try {
                this.finalize();
            } catch (Throwable throwable) {
                throwable.printStackTrace();
            }
        } else {
            gamepanel = new GamePanel(this);
            setContentView(gamepanel);

	        if (extras.getBoolean("l")){
		        if (!gamepanel.loadState()){
			        this.finish();
		        }
	        } else {
		        Grid grid = (Grid) extras.getSerializable("g");
		        gamepanel.setNewGrid(grid);
		        gamepanel.postInvalidate();
	        }
        }
    }

    @Override
    protected void onPause() {
        gamepanel.saveState();
        super.onPause();
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (keyCode==KeyEvent.KEYCODE_MENU){
            Intent intent = new Intent(this, SettingsActivity.class);
            startActivity(intent);
        }
        return super.onKeyDown(keyCode, event);
    }

    @Override
    public void onBackPressed() {
        Intent intent = new Intent(this, MenuActivity.class);
        startActivity(intent);
	    overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
        finish();
    }
}

File: app/src/main/java/org/oar/minesweeper/GameActivity.kt
package org.oar.minesweeper

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.*
import android.view.ViewTreeObserver.OnGlobalLayoutListener
import org.oar.minesweeper.models.Grid
import org.oar.minesweeper.models.GridGenerationDetails
import org.oar.minesweeper.ui.views.GameView
import org.oar.minesweeper.utils.ActivityUtils.animateStartActivity


/**
 * An example full-screen activity that shows and hides the system UI (i.e.
 * status bar and navigation/system bar) with user interaction.
 */
class GameActivity : Activity() {
    private val gameView: GameView by lazy { findViewById(R.id.panel) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)

        val extras = this.intent.extras

        if (extras == null) {
            finish();
        } else {
            setContentView(R.layout.activity_playing_grid)

            window.insetsController?.apply {
                hide(WindowInsets.Type.statusBars())
                hide(WindowInsets.Type.navigationBars())
            }

            if (extras.getBoolean("l")) {
                if (!gameView.loadState()) {
                    finish()
                }
            } else {
                val grid = extras.getSerializable("grid") as Grid
                val details = extras.getSerializable("options") as GridGenerationDetails

                gameView.viewTreeObserver.addOnGlobalLayoutListener(
                    object: OnGlobalLayoutListener {
                        override fun onGlobalLayout() {
                            // at this point, view dimensions are set (view.width and view.height)
                            gameView.setGrid(grid, details)
                            gameView.postInvalidate()
                            // make sure it is only executed once:
                            gameView.viewTreeObserver.removeOnGlobalLayoutListener(this)
                        }
                    }
                )
            }
        }
    }

    override fun onPause() {
        gameView.saveState()
        super.onPause()
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        if (keyCode == KeyEvent.KEYCODE_MENU) {
            val intent = Intent(this, SettingsActivity::class.java)
            startActivity(intent)
        }
        return super.onKeyDown(keyCode, event)
    }

    override fun onBackPressed() {
        val intent = Intent(this, MenuActivity::class.java)
        animateStartActivity(intent, true)
    }
}